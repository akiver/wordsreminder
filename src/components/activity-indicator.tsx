import React from 'react';
import { ActivityIndicator as RNActivityIndicator, StyleSheet, ActivityIndicatorProps, ViewStyle } from 'react-native';
import { useTheme } from '@theme/use-theme';

type Props = ActivityIndicatorProps;

export function ActivityIndicator({ size = 'small' }: Props) {
  const theme = useTheme();

  return (
    <RNActivityIndicator
      color={theme.primary025}
      size={size}
      style={styles.activityIndicator}
      accessibilityHint="Loading"
    />
  );
}

type Style = {
  activityIndicator: ViewStyle;
};

const styles = StyleSheet.create<Style>({
  activityIndicator: {
    padding: 10,
  },
});
