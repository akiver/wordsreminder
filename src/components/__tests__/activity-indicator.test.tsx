import React from 'react';
import { render } from '@testing-library/react-native';
import { ActivityIndicator } from '../activity-indicator';

describe('ActivityIndicator', () => {
  it('should match snapshot', () => {
    const tree = render(<ActivityIndicator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
