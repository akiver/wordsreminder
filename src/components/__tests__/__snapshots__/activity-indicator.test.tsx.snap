// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityIndicator should match snapshot 1`] = `
<ActivityIndicator
  accessibilityHint="Loading"
  color="#ffffff"
  size="small"
  style={
    Object {
      "padding": 10,
    }
  }
/>
`;
